File: elixir/etl.ex
defmodule ETL do
  @doc """
  Transforms an old Scrabble score system to a new one.

  ## Examples

    iex> ETL.transform(%{1 => ["A", "E"], 2 => ["D", "G"]})
    %{"a" => 1, "d" => 2, "e" => 1, "g" => 2}
  """
  @spec transform(map) :: map
  def transform(input) do
    input
    |> Enum.reduce(%{}, fn {n, lst}, acc ->
      lst
      |> Enum.reduce(acc, fn c, acc ->
        Map.put(acc, String.downcase(c), n)
      end)
    end)
  end
end

File: go/etl.go
package etl

import "strings"

func Transform(in map[int][]string) map[string]int {
	mp := map[string]int{}
	for k, v := range in {
		for _, c := range v {
			mp[strings.ToLower(c)] = k
		}
	}
	return mp
}
